### DISCUSSION TITLE
Can someone tell why Gatsby SSG is faster than Next.js SSG?

### DISCUSSION THREAD

**Discussion by baymac** [gatsbyjs/gatsby#32551] [Help] [answered]
I am curious 🙄

**Comment by herecydev** [comment, marked as answer]
Faster in what regard? From the user's perspective they are downloading a pre-rendered HTML document that gets hydrated with a react application.

**Reply by baymac** [reply]
From my experience, Gatsby pages load faster compared to Next.js SSG pages. Although I did the experiment on 2 relatively different websites I had created and a few more sites I came across. After your comment, I compared Next.js blog starter with Gatsby blog starter in Lighthouse, both have more or less the same results.

Gatsby starter:
<img width="714" alt="Screenshot 2021-07-31 at 2 54 30 PM" src="https://user-images.githubusercontent.com/23079344/127736629-2ed85e72-e9d2-4c6e-9dd8-8dcd7d0ccfbc.png">

Nextjs starter:
<img width="714" alt="Screenshot 2021-07-31 at 3 42 40 PM" src="https://user-images.githubusercontent.com/23079344/127736658-2daf341d-905f-4587-87ff-03bd6559eacb.png">

Although there is slightly more CLS and TTI in Gatsby. How does Gatsby SSG work different from Next.js?

**Reply by herecydev** [reply]
You can't compare that way. Create a page in both next and gatsby that just has the text "Hello world". It will have 0 CLS in both.

TTI will be influenced by the framework (more heavy javascript == longer TTI), Gatsby and next will handle routing and other stuff and will pull down a different amount of javascript to deal with that. Honestly, it's going to be negligable. Both will do a great job of handling SSG.

**Reply by baymac** [reply]
Cool man, thanks. Looks like apart from data fetching methods, both SSGs work the same like you described for Next.js. One difference I know is Gatsby inlines imp CSS which perhaps make site loading seem faster to user.
